packaging/greenboot/functions.sh

```diff
@@ -221,3 +237,43 @@ function print_failure_logs() {
         fi
     done
 }
+
+# Run a command specified in the arguments, redirect its output to a temporary
+# file and add this file to 'LOG_FAILURE_FILES' setting so that is it printed
+# in the logs if the script exits with failure.
+#
+# All the command output including stdout and stderr is redirected to its log file.
+#
+# arg1: A name to be used when creating "/tmp/${name}.XXXXXXXXXX" temporary files
+# arg2: A command to be run
+# return: None
+function log_failure_cmd() {
+    local -r logName="$1"
+    local -r logCmd="$2"
+    local -r logFile=$(mktemp "/tmp/${logName}.XXXXXXXXXX")
+
+    # Run the command ignoring errors and log its output
+    (${logCmd}) &> "${logFile}" || true
+    # Save the log file name in the list to be printed
+    LOG_FAILURE_FILES+=("${logFile}")
+}
+
+# The script exit handler logging the FAILURE or FINISHED message depending
+# on the exit status of the last command. If the 'LOG_POD_EVENTS' environment
+# variable is set, also log the list of pods and their events on failure.
+#
+# args: None
+# env: 'LOG_POD_EVENTS' environment variable to enable pod-specific logging
+# return: None
+function log_script_exit() {
+    if [ "$?" -ne 0 ] ; then
+        if ${LOG_POD_EVENTS}; then
+            log_failure_cmd "pod-list" "${OCGET_CMD} pods -A -o wide"
+            log_failure_cmd "pod-events" "${OCGET_CMD} events -A"
```

Comment:
This is outside of scope of this PR, but what do you think about sorting events by time? Currently it's sorted by namespace and it's not helping with quickly seeing what happened last.
```--sort-by='.metadata.creationTimestamp'`